ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -192,6 +205,31 @@ public Response handleKsqlStatements(final KsqlRequest request) {
     return Response.ok(result).build();
   }
 
+  private void waitForCommandOffset(final Optional<Long> commandOffset) {
+    if (commandOffset.isPresent()) {
+      final long offset = Objects.requireNonNull(commandOffset.get(), "commandOffset is null");
+      final CompletableFuture<Void> future =
+          replayableCommandQueue.getConsumerPositionFuture(offset);
+
+      try {
+        future.get(distributedCommandResponseTimeout, TimeUnit.MILLISECONDS);
+      } catch (final ExecutionException e) {
+        throw new RuntimeException(
+            "Error waiting for command offset of " + String.valueOf(offset), e.getCause());
+      } catch (final InterruptedException e) {
+        throw new RuntimeException(
+            "Interrupted while waiting for command offset of " + String.valueOf(offset), e);
+      } catch (final TimeoutException e) {
+        throw new RuntimeException(
```

Comment:
By throwing RuntimeException here, we will return a generic 500 error. We should return a specific error class for this condition, with the appropriate HTTP code. 

I'd say the appropriate error code would be a 503 Service Unavailable.

We also need to return a specific error indicating that we timed out waiting for the server to catch up on the command topic. Check out io.confluent.ksql.rest.server.resources.Errors. You can define a new error code like:

```
public static final int ERROR_CODE_COMMAND_QUEUE_CATCHUP_TIMEOUT = toErrorCode(toErrorCode(SERVICE_UNAVAILABLE.getStatusCode()) + 1);

public static Response commandQueueCatchupTimeout(final String msg) {
    return Response
        .status(SERVICE_UNAVAILABLE)
        .entity(new KsqlErrorMessage(ERROR_CODE_COMMAND_QUEUE_CATCHUP_TIMEOUT, msg))
        .build();
  }
```